Jarvis-Platform/jarvis-studio
# Feature/Users content update

[comment]: [![Status](https://img.shields.io/badge/Status-DEVELOPMENT-orange.svg)]()
[comment]: [![Status](https://img.shields.io/badge/Status-HOLD-inactive.svg)]()

[![Status](https://img.shields.io/badge/Status-READY-brightgreen.svg)]()


## Description
A few sentences describing the overall goals of the pull request's commits.


## Todos
### General
- [x] Convert view to TypeScript

### UI
- [x] Add alert for `disabled` accounts
- [x] Add alert for non verified email on account
- [x] Rearrange form inputs

### Create
- [x] Use cloud function & push to table
- [x] Add snackBar for validation

### Delete
- [x] Cloud function
- [x] Add modal for confirmation
- [x] Add snackBar for validation

### Archive
- [x] Add archive action
- [x] Add modal for confirmation
- [x] Add snackBar for validation


## Deploy Notes
Review & merge


## Steps to Test or Reproduce
```sh
git checkout master
git pull
git checkout feature/delete-user
```

1. Try to create a user => it should be ok.
1. Try to delete a user => it should be ok.
1. Try to archive a user => it should be ok.


## Impacted Areas in Application
List general components of the application that this PR will affect:
- `src/components/settings/UsersContent.vue`
- `functions/`
